src/Illuminate/Auth/Access/Gate.php

```diff
@@ -113,6 +120,9 @@ public function define($ability, $callback)
             $this->abilities[$ability] = $callback;
         } elseif (is_string($callback)) {
             $this->abilities[$ability] = $this->buildAbilityCallback($ability, $callback);
+            if ($this->abilities[$ability] instanceof \Closure) {
```

Comment:
Why do you need to check for this here? Doesn't `buildAbilityCallback` always return a Closure? I'm still a little unsure why you need to maintain this separate list of Closure callbacks.